repository/src/main/java/tech/ydb/yoj/repository/db/cache/TransactionLocal.java

```diff
@@ -3,46 +3,48 @@
 import lombok.NonNull;
 import tech.ydb.yoj.repository.BaseDb;
 import tech.ydb.yoj.repository.db.TxOptions;
+import tech.ydb.yoj.repository.db.projection.MigrationProjectionCache;
 import tech.ydb.yoj.repository.db.projection.ProjectionCache;
 import tech.ydb.yoj.repository.db.projection.RoProjectionCache;
 import tech.ydb.yoj.repository.db.projection.RwProjectionCache;
 
-import java.util.IdentityHashMap;
-import java.util.Map;
-import java.util.function.Supplier;
-
 public class TransactionLocal {
-    private final Map<Supplier<?>, Object> singletons = new IdentityHashMap<>();
```

Comment:
I think that this singletons is made for case when in tx wasn't any query. But this objects is lightwave, maybe we can do it without singletons? Else I have to do some painful refactoring for this feature ;(